modules/api/pkg/ee/clusterbackup/storage-location/provider.go

```diff
@@ -110,57 +141,78 @@ func (p *BackupStorageProvider) CreateUnsecured(ctx context.Context, cbslName, p
 	}
-	if err := p.privilegedClient.Create(ctx, cbsl); err != nil {
+	if err := client.Create(ctx, cbsl); err != nil {
+		return nil, common.KubernetesErrorToHTTPError(err)
+	}
+	ownerReference := metav1.OwnerReference{
+		APIVersion: ownerReferenceApi,
+		Kind:       ownerReferenceKind,
+		Name:       cbslFullName,
+		UID:        cbsl.UID,
+	}
+
+	secret.ObjectMeta.OwnerReferences = append(secret.ObjectMeta.OwnerReferences, ownerReference)
+
+	if err := p.privilegedClient.Create(ctx, secret); err != nil {
 		return nil, err
 	}
 	return cbsl, nil
 }
 
-func (p *BackupStorageProvider) DeleteUnsecured(ctx context.Context, name string) error {
+func (p *BackupStorageProvider) Delete(ctx context.Context, userInfo *provider.UserInfo, name string) error {
+	if userInfo == nil {
+		return errors.New("a user is missing but required")
+	}
+
+	client := p.privilegedClient
+
 	cbsl := &kubermaticv1.ClusterBackupStorageLocation{}
-	if err := p.privilegedClient.Get(ctx, types.NamespacedName{Name: name, Namespace: resources.KubermaticNamespace}, cbsl); err != nil {
+	if err := client.Get(ctx, types.NamespacedName{Name: name, Namespace: resources.KubermaticNamespace}, cbsl); err != nil {
 		if apierrors.IsNotFound(err) {
 			return nil
 		}
 		return err
 	}
 
-	if cbsl.Spec.Credential != nil {
-		secretName := cbsl.Spec.Credential.Name
-		if err := p.deleteCredentials(ctx, secretName); err != nil {
+	if !userInfo.IsAdmin {
+		var err error
+		client, err = p.getImpersonatedClient(userInfo)
+		if err != nil {
 			return err
 		}
 	}
-	return p.privilegedClient.Delete(ctx, cbsl)
-}
 
-func (p *BackupStorageProvider) deleteCredentials(ctx context.Context, secretName string) error {
-	secret := &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      secretName,
-			Namespace: resources.KubermaticNamespace,
-		},
-	}
-	if err := p.privilegedClient.Delete(ctx, secret); err != nil && !apierrors.IsNotFound(err) {
+	if err := client.Delete(ctx, cbsl); err != nil {
 		return err
 	}
 	return nil
 }
 
-func (p *BackupStorageProvider) PatchUnsecured(ctx context.Context, cbslName string, cbsl *kubermaticv1.ClusterBackupStorageLocation, updatedCreds apiv2.S3BackupCredentials) (*kubermaticv1.ClusterBackupStorageLocation, error) {
+func (p *BackupStorageProvider) Patch(ctx context.Context, userInfo *provider.UserInfo, cbslName string, cbsl *kubermaticv1.ClusterBackupStorageLocation, updatedCreds apiv2.S3BackupCredentials) (*kubermaticv1.ClusterBackupStorageLocation, error) {
+	if userInfo == nil {
+		return nil, errors.New("a user is missing but required")
+	}
+
+	client := p.privilegedClient
+
 	existing := &kubermaticv1.ClusterBackupStorageLocation{}
-	if err := p.privilegedClient.Get(ctx, types.NamespacedName{Name: cbslName, Namespace: resources.KubermaticNamespace}, existing); err != nil {
+	if err := client.Get(ctx, types.NamespacedName{Name: cbslName, Namespace: resources.KubermaticNamespace}, existing); err != nil {
 		return nil, err
 	}
```

Comment:
same here. Is the priviledged client required, or could we just use the impersonated client for getting it?

Reply:
done